Src/Support/Google.Apis/Upload/ResumableUpload.cs

```diff
@@ -817,6 +778,93 @@ private string GetContentRangeHeader(long chunkStart, long chunkSize)
             }
         }
 
+        /// <summary>
+        /// A buffer to be uploaded. This abstraction will eventually avoid the requirement for large byte arrays
+        /// which might end up on the large object heap.
+        /// </summary>
+        private class UploadBuffer
+        {
+            /// <summary>
+            /// The amount of usable data within the buffer.
+            /// </summary>
+            public int Length { get; private set; }
+
+            /// <summary>
+            /// The data in the buffer.
+            /// </summary>
+            private readonly byte[] data;
+
+            public UploadBuffer(int capacity)
```

Comment:
I would pass (Upload)ChunkSize here and remove it from the parameters of the methods.
As it is now, MoveUnsentData... needs to receive the same ChunkSize as the last CreateContent. I don't think ChunkSize is going to change anyway, and I think the implementation would be clearer.

Reply:
Hmm... not sure. While I agree that it won't change, I'm not sure that it's logically "state of the buffer". What might make more sense would be to keep a `ResumeableUpload parent` field that we can use for *anything* like ChunkSize, BufferSize etc.